examples/private_information_retrieval/module/rust/src/database.rs

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Private Information Retrieval example for Project Oak.
+//!
+//! This shows how an Oak Node can aggregate data samples and report aggregated values if there are
+//! enough samples to hide individual contributors (enforces k-anonymity).
+//!
+//! Clients invoke the module by providing data samples that contain a bucket ID
+//! and a Sparse Vector - a dictionary with integer keys.
+
+use crate::proto::{Location, PointOfInterest};
+use log::{debug, error};
+use oak::OakError;
+use oak_abi::proto::oak::application::ConfigMap;
+use quick_xml::de::from_str;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Database {
+    #[serde(rename = "lastUpdate", default)]
+    last_update: String,
+    version: String,
+    #[serde(rename = "station", default)]
+    stations: Vec<Station>,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Station {
+    id: u32,
+    name: String,
+    #[serde(rename = "terminalName", default)]
+    terminal_name: String,
+    lat: f32,
+    long: f32,
+    installed: bool,
+    locked: bool,
+    #[serde(rename = "installDate", default)]
+    install_date: String,
+    #[serde(rename = "removalDate", default)]
+    removal_date: String,
+    temporary: bool,
+    #[serde(rename = "nbBikes", default)]
```

Comment:
Where does all this structure come from? If it's documented somewhere on the internet, please link to it. If not, please explain how you arrived to it.